python/lsst/meas/base/forcedPhotCcd.py

```diff
@@ -130,8 +130,21 @@ def fetchReferences(self, dataRef, exposure):
         All work is delegated to the references subtask; see CoaddSrcReferencesTask for information
         about the default behavior.
         """
-        bbox = exposure.getBBox()
-        return self.references.fetchInBox(dataRef, bbox, exposure.getWcs())
+        references = lsst.afw.table.SourceCatalog(self.references.schema)
+        badParents = set()
+        unfiltered = self.references.fetchInBox(dataRef, exposure.getBBox(), exposure.getWcs())
+        for record in unfiltered:
+            if record.getFootprint() is None or record.getFootprint().getArea() == 0:
+                if record.getParent() != 0:
+                    self.log.warn("Skipping reference %s (child of %s) with bad Footprint" %
+                    (record.getId(), record.getParent()))
+                else:
+                    self.log.warn("Skipping reference parent %s with bad Footprint" % (record.getId(),))
+                    badParents.add(record.getId())
+            elif record.getParent() not in badParents:
+                references.append(record)
+        references.sort()   # need to ensure catalog is in ID order so find methods work
+        return references
```

Comment:
With this change, the documentation at the top of this method is a bit imprecise: we're not returning just an iterable, we're returning a `SourceCatalog`.  And it's probably a good idea to mention in those docs that it's sorted by ID and guarantees that all included children have their parent included and all `Footprints` are valid.